src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -243,6 +253,15 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             if (mode == Mode.GlobbedHref || mode == Mode.Fallback && !string.IsNullOrEmpty(HrefInclude))
             {
+                if (TryResolveUrl(HrefInclude, tryEncodeApplicationPath: false, resolvedUrl: out resolvedUrl))
```

Comment:
@NTaylorMullen why open a new issue instead of fixing the code as part of this PR? As you said offline, the implementation would be simpler.